.internal-ci/util/bump_cargo_version.sh

```diff
@@ -0,0 +1,121 @@
+    echo "    --current-tag - current tag defined in Cargo.toml"
+    echo "        as a safeguard against updating packages that are not in sync with the repo version"
+    echo "    --new-tag - new tag to replace current"
+    echo "    --author - (Optional) author= string to match. Default 'MobileCoin'"
+    echo "    --dry-run - (Optional) parse files and show error/actions, but don't change versions"
+    echo ""
+    echo "Example:"
+    echo "${0} --current-tag 1.2.2 --new-tag 1.2.3"
+    echo ""
+}
+
+is_set()
+{
+    var_name="${1}"
+    if [ -z "${!var_name}" ]
+    then
+        echo "${var_name} is not set."
+        usage
+        exit 1
+    fi
+}
+
+while (( "$#" ))
+do
+    case "${1}" in
+        --help | -h)
+            usage
+            exit 0
+            ;;
+        --current-tag)
+            current_tag="${2}"
+            shift 2
+            ;;
+        --new-tag)
+            new_tag="${2}"
+            shift 2
+            ;;
+        --author)
+            author="${2}"
+            shift 2
+            ;;
+        --dry-run)
+            dry_run="true"
+            shift
+            ;;
+        *)
+            echo "${1} unknown option"
+            usage
+            exit 1
+            ;;
+    esac
+done
+
+if ! which dasel >/dev/null 2>&1
+then
+    echo "This script requires dasel https://github.com/TomWright/dasel for parsing toml. Please install it in your PATH"
+    exit 1
+fi
+
+is_set current_tag
+is_set new_tag
+
+# set author
+author=${author:-MobileCoin}
+
+# Get list of Cargo.toml files, exclude ./cargo and ./target directories
+files=$(find . \( -path ./cargo -o -path ./target \) -prune -o -name Cargo.toml -print)
+
+for f in ${files}
+do
+    echo "-- Checking File: ${f}"
+    # does the authors block exist?
+    if dasel get -f "${f}" -m --plain -s '.package.authors.[*]' >/dev/null 2>&1
+    then
+        # Get authors for package and make sure the list matches --author
+        authors=$(dasel get -f "${f}" -m --plain -s '.package.authors.[*]')
+        if [[ "${authors}" == "${author}" ]]
+        then
+            # Check to make sure current package version matches current_tag
+            p_version=$(dasel get -f "${f}" -m --plain -s '.package.version')
+            if [[ "${p_version}" == "${current_tag}" ]]
+            then
+                # I would use dasel here, but the output file ends up with properties in alpha order. this will make a mess :-(
+                # so punt and use sed.
+                if [[ -n "${dry_run}" ]]
+                then
+                    echo "sed -i -e \"s/version = \\\"${current_tag}\\\"/version = \\\"${new_tag}\\\"/g\" ${f}"
+                else
+                    sed -i -e "s/version = \"${current_tag}\"/version = \"${new_tag}\"/g" "${f}"
+                fi
+            else
+                echo "!! File current version ${p_version} doesn't match --current-tag ${current_tag}"
```

Comment:
nit: Should this say `skipping` like the other checks?
```suggestion
                echo "!! File current version ${p_version} doesn't match --current-tag ${current_tag}" - skipping
```